zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/CreateDeploymentMultiplePartitionsTest.java

```diff
@@ -534,6 +534,33 @@ public void shouldWriteDrgAndDecisionCreatedEventsWithSameKeys() {
         .hasSize(1);
   }
 
+  @Test
+  public void shouldWriteDecisionCreatedEventsWithDeploymentKeyOnAllPartitions() {
+    // given
+    final var decisionId = "jedi_or_sith";
+
+    // when
+    final var deployment =
+        ENGINE.deployment().withXmlClasspathResource(DMN_DECISION_TABLE_V2).deploy();
+
+    // then
+    ENGINE.forEachPartition(
+        partitionId -> {
+          final var record =
+              RecordingExporter.decisionRecords()
+                  .withPartitionId(partitionId)
+                  .withIntents(DecisionIntent.CREATED)
+                  .withDecisionId(decisionId)
+                  .limit(1)
+                  .getFirst();
```

Comment:
🔧 Nit: no need to limit to 1 if we're getting the first. `getFirst()` would already short circuit if an entry is available.